### ISSUE TITLE
Support Linear Fused Batchnorm

### ISSUE THREAD

**Issue by mcr229** [pytorch/executorch#11587]
### 🚀 The feature, motivation and pitch

Currently XNNPACK only supports batchnorm when it succeeds a convolution. This is because we can fuse it with the preceeding convolution operation. However we can also do this for Linear layers, by fusing batchnorm with the previous linear. Take a look at the code for fusing with conv, and try to add a pass to do the same for linear:

https://github.com/pytorch/executorch/blob/main/backends/xnnpack/_passes/fuse_batch_norm_with_conv.py

**Comment by keyprocedure** [comment]
I’d like to take this on

**Comment by mcr229** [comment]
assigning it to you!

**Comment by keyprocedure** [comment]
There's quite a bit of duplicated logic between [fuse_batch_norm_with_linear.py](https://github.com/pytorch/executorch/pull/11805/files#diff-cc04f0c24ffa98dec2b077ffc8dd3bacbd12c744dd5e31cae209a2575408f02a) and [fuse_batch_norm_with_conv.py](https://github.com/pytorch/executorch/blob/main/backends/xnnpack/_passes/fuse_batch_norm_with_conv.py). 
Would it be better to refactor it in this PR, or follow up with another PR after the current one lands?
